scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -249,6 +352,28 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
     )
   }
 
+  test("Merge (++) two pipes") {
+    val a = TypedPipe.from(Seq(5, 3, 2, 6, 1, 4))
+    val b = TypedPipe.from(Seq(15, 13, 12, 16, 11, 14))
+
+    beamMatchesSeq(
+      a ++ b,
+      Seq(1, 2, 3, 4, 5, 6, 11, 12, 13, 14, 15, 16)
+    )
+  }
+
+  test("Merge (++) many pipes") {
```

Comment:
testing a `a ++ a` example is interesting since it should be optimized into a `flatMap` that just duplicates each element. Same with `a ++ a ++ a` I think.

Reply:
for sure

Reply:
I added a test to check merging a pipe to itself produces the right result but had a hard time trying to test the shape of the conversion, in a way that includes the optimizations. Do you think we can leave that work to a separate PR? I don't want to grow this one too much more.